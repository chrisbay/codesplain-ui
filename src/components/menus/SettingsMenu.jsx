import React, { PropTypes } from 'react';
import {
  IconButton,
  IconMenu,
} from 'material-ui';
import SettingsIcon from 'material-ui/svg-icons/action/settings';

import DeleteMenu from './DeleteMenu';
import KeymapPicker from './KeymapPicker';
import ThemePicker from './ThemePicker';

const SETTINGS_ICON_BUTTON = <IconButton><SettingsIcon /></IconButton>;

const SettingsMenu = (props) => {
  const {
    codeMirrorTheme,
    deleteEnabled,
    onDeleteClick,
    onKeymapChange,
    onThemeChange,
    selectedKeymap,
  } = props;
  return (
    <IconMenu
      iconButtonElement={SETTINGS_ICON_BUTTON}
    >
      <ThemePicker
        codeMirrorTheme={codeMirrorTheme}
        onChange={onThemeChange}
      />
      <KeymapPicker
        selectedKeymap={selectedKeymap}
        onChange={onKeymapChange}
      />
      { deleteEnabled ? <DeleteMenu onDeleteClick={onDeleteClick} /> : null }
    </IconMenu>
  );
};

SettingsMenu.propTypes = {
  codeMirrorTheme: PropTypes.string,
  deleteEnabled: PropTypes.bool.isRequired,
  onDeleteClick: PropTypes.func.isRequired,
  onKeymapChange: PropTypes.func.isRequired,
  onThemeChange: PropTypes.func.isRequired,
  selectedKeymap: PropTypes.string,
};

SettingsMenu.defaultProps = {
  codeMirrorTheme: 'codesplain',
  selectedKeymap: 'default',
};

export default SettingsMenu;
