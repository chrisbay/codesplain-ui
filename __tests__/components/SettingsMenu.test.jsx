import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import SettingsMenu from '../../src/components/menus/SettingsMenu';

const defaultProps = {
  deleteEnabled: false,
  onDeleteClick: jest.fn(),
  onKeymapChange: jest.fn(),
  onThemeChange: jest.fn(),
};

describe('<SettingsMenu />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches snapshot of when user cannot delete snippet', () => {
      const wrapper = shallowWithContext(
        <SettingsMenu {...defaultProps} />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('matches snapshot of when user can delete snippet', () => {
      const wrapper = shallowWithContext(
        <SettingsMenu
          {...defaultProps}
          deleteEnabled
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('prop: onThemeChange', () => {
    it('is forwarded to ThemePicker', () => {
      const onThemeChange = () => {};
      const wrapper = shallowWithContext(
        <SettingsMenu
          {...defaultProps}
          onThemeChange={onThemeChange}
        />,
      );
      const themePicker = wrapper.find('ThemePicker');
      expect(themePicker.prop('onChange')).toEqual(onThemeChange);
    });
  });

  describe('prop: onKeymapChange', () => {
    it('is forwarded to KeymapPicker', () => {
      const onKeymapChange = () => {};
      const wrapper = shallowWithContext(
        <SettingsMenu
          {...defaultProps}
          onKeymapChange={onKeymapChange}
        />,
      );
      const keymapPicker = wrapper.find('KeymapPicker');
      expect(keymapPicker.prop('onChange')).toEqual(onKeymapChange);
    });
  });

  describe('prop: selectedKeymap', () => {
    it('is forwarded to KeymapPicker', () => {
      const selectedKeymap = 'monokai';
      const wrapper = shallowWithContext(
        <SettingsMenu
          {...defaultProps}
          selectedKeymap={selectedKeymap}
        />,
      );
      const keymapPicker = wrapper.find('KeymapPicker');
      expect(keymapPicker.prop('selectedKeymap')).toEqual(selectedKeymap);
    });
  });
});
