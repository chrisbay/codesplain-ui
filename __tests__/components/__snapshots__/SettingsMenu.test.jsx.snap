exports[`<SettingsMenu /> snapshot tests matches snapshot of when user can delete snippet 1`] = `
<IconMenu
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  animated={true}
  iconButtonElement={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}>
      <ActionSettings />
    </IconButton>
  }
  multiple={false}
  onItemTouchTap={[Function]}
  onKeyboardFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onRequestChange={[Function]}
  onTouchTap={[Function]}
  open={null}
  targetOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  touchTapCloseDelay={200}
  useLayerForClickAway={false}>
  <ThemePicker
    codeMirrorTheme="codesplain"
    onChange={[Function]} />
  <KeymapPicker
    onChange={[Function]}
    selectedKeymap="default" />
  <DeleteMenu
    onDeleteClick={[Function]} />
</IconMenu>
`;

exports[`<SettingsMenu /> snapshot tests matches snapshot of when user cannot delete snippet 1`] = `
<IconMenu
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  animated={true}
  iconButtonElement={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}>
      <ActionSettings />
    </IconButton>
  }
  multiple={false}
  onItemTouchTap={[Function]}
  onKeyboardFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onRequestChange={[Function]}
  onTouchTap={[Function]}
  open={null}
  targetOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  touchTapCloseDelay={200}
  useLayerForClickAway={false}>
  <ThemePicker
    codeMirrorTheme="codesplain"
    onChange={[Function]} />
  <KeymapPicker
    onChange={[Function]}
    selectedKeymap="default" />
</IconMenu>
`;
